### PR TITLE
[core] Fixes #777 Enable the linter deadcode of  gometalinter

### LINKED ISSUE #777: Add deadcode linter and fix errors.
gometalinter --disable-all --vendor --enable deadcode ./...
cmd/skycoin/skycoin.go:752:1:warning: InitTransaction is unused (deadcode)
src/api/webrpc/webrpc.go:19:1:warning: errMsgInvalidRequest is unused (deadcode)
src/daemon/storage.go:17:1:warning: matchFunc is unused (deadcode)
src/gui/blockchain.go:18:1:warning: lastBlockNum is unused (deadcode)
src/util/logging/logging.go:34:1:warning: levelNames is unused (deadcode)
make: *** [lint] Error 1

### PR DESCRIPTION
**Pull request by stdevHan**
Fixes #777

Changes:
-Change in `Makefile` by adding the `-E deadcode` parameter to `gometalinter` and correcting the problems .

Does this change need to mentioned in CHANGELOG.md?

### CODE REVIEW

**Comment by gz-c** [comment]
merge conflict

**Comment by stdevStark** [comment]
I'll take a look .

**Comment by stdevStark** [comment]
done ... wait for the Travis results.

File: src/api/integration/integration_test.go [hunk 1]
```diff
@@ -3699,32 +3687,6 @@ func getAddressBalance(t *testing.T, c *api.Client, addr string) (uint64, uint64
 	return bp.Confirmed.Coins, bp.Confirmed.Hours
 }
 
-func checkNoSensitiveData(t *testing.T, w *wallet.Wallet) {
```
[the remark this answers is not in the document]
Comment (iketheadore) [reply src/api/integration/integration_test.go]: Yes, it's not used, we do sensitive data check in the EncryptWallet test directly, but forget to check the entries, I will send a PR to fix that.

**Pull request merged**